### DISCUSSION TITLE
How to define default colors for toggle, bang etc

### DISCUSSION THREAD

**Discussion by mk-archaeo** [sebshader/pdnext#14] [Q&A] [answered]
Hi,
I don't know if it is possible to define default colors in the tcl-file for toggle, bang, number2, for the sliders and radios.
What are the category names for these? If this is implemented.

Thanks for your great work.

Marco

**Comment by sebshader** [comment, marked as answer]
hi @mk-archaeo
There aren't any hooks for these "iemgui" objects. The issue is that people set and save specific color values for these objects when distributing GOP abstractions and patches. So the colors for these objects are set and saved "per-object" in the properties.
The only possible hooks I've been thinking about putting in is for when the objects are created. Then you could define default properties when an object was first created, but not when loading from a file. However this might still be problematic to implement since I'm not sure if it will be possible to tell the difference between when an object is first created and when it is loaded from a patch file.
edit: I might consider also creating a hook for gui objects created in the root canvas (one that has no parents) even if already set

**Reply by sebshader** [reply]
In addition if you create the object with a certain set of colors and then change the theme the object wouldn't change to reflect the new theme

**Reply by mk-archaeo** [reply]
yes, you are right.

**Comment by mk-archaeo** [comment]
Thank you for the answer. Didn't know they are something extra. I understand the problematic. I'm no programming expert, maybe is it possible something like [loadbang] in the patches, so a default color is set, but if something is set in the properties it overrides the default, don't know. The idea with the root canvas is a interesting idea. 

Thanks again for your time and your effort

Marco

**Comment by burein-ita** [comment]
Would it be possible to only allow changing the color of the iemgui outlines AFAIK it's not possible to set those through the "Properties" dialog, so it shouldn't break compatibility.

**Reply by sebshader** [reply]
maybe.. in GOP where iemgui background color and canvas are explicitly set to similar light colors and the user also sets the outline to a light color it might be hard to see but that's the only case I can think of where it wouldn't work with other patches.

but I suppose certain settings for gatoms are the same way, it could be another 'use/set at your own loss of utility' type thing

**Reply by sebshader** [reply]
another aspect is that the colors feature is really supposed to be based on the equivalent PR for Pd, and I'm not sure they would accept such a variable because it isn't as consistent with pd overall (the thing to do would be to add it as another setting for iemguis)
But I might add it even if they don't.
